lib/ohai/plugins/packages.rb

```diff
@@ -47,7 +47,12 @@
 
       pkgs.each do |pkg|
         name, epoch, version, release, installdate, arch = pkg.split
-        packages[name] = { "epoch" => epoch, "version" => version, "release" => release, "installdate" => installdate, "arch" => arch }
+        unless packages[name].nil?
```

Comment:
Ah, that was to help make the determination if I'm dealing with a duplicate or not - only a duplicate would trigger adding additional items for that software package.  I'll add comments there, and I'm definitely open to suggestion on the logic there if there's a better way to handle a duplicate situation.

With RPM, it seems that we only get the last listed duplicate.  I'm not sure if there's a good way to determine what that's going to be on any given system as it's completely dependent on rpm's output ordering and the exist logic just overwrites previous entries as it runs through the list.  

I'll look at what I can do to include valid tests as you're right and we don't want to regress.  Other than kernel related packages, I doubt many others can actually have multiple copies at any given time unless perhaps talking source rpms which aren't likely to exist on a chef-managed server though I think this would catch that..